### ISSUE TITLE
byte index is not a char boundary

### ISSUE THREAD

**Issue by MFukazawa** [biomejs/biome#4205]
### Environment information

```bash
This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: crates/biome_js_factory/src/utils.rs:34:36
Thread Name: biome::worker_1
Message: byte index 3 is not a char boundary; it is inside '⏱' (bytes 2..5) of `%c⏱ `

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: crates/biome_js_factory/src/utils.rs:34:36
Thread Name: biome::worker_10
Message: byte index 1 is not a char boundary; it is inside '°' (bytes 0..2) of `°, `

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: crates/biome_js_factory/src/utils.rs:34:36
Thread Name: biome::worker_10
Message: byte index 3 is not a char boundary; it is inside '⏱' (bytes 2..5) of `%c⏱ `

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: crates/biome_js_factory/src/utils.rs:34:36
Thread Name: biome::worker_11
Message: byte index 1 is not a char boundary; it is inside '°' (bytes 0..2) of `°, `

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: crates/biome_js_factory/src/utils.rs:34:36
Thread Name: biome::worker_11
Message: byte index 3 is not a char boundary; it is inside '⏱' (bytes 2..5) of `%c⏱ `

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: crates/biome_js_factory/src/utils.rs:34:36
Thread Name: biome::worker_1
Message: byte index 1 is not a char boundary; it is inside '°' (bytes 0..2) of `°, `
```

## Rage Output
```
CLI:
  Version:                      1.9.3
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v18.17.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/8.6.12"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Linter:
  JavaScript enabled:           true
  JSON enabled:                 true
  CSS enabled:                  true
  GraphQL enabled:              false
  Recommended:                  false
  All:                          false
  Enabled rules:
  suspicious/noCatchAssign
  suspicious/noUnsafeNegation
  complexity/noMultipleSpacesInRegularExpressionLiterals
  a11y/useValidLang
  suspicious/useValidTypeof
  a11y/useValidAriaRole
  correctness/noConstantCondition
  suspicious/noDuplicateParameters
  a11y/useAriaActivedescendantWithTabindex
  suspicious/noAssignInExpressions
  style/useDefaultParameterLast
  correctness/noConstructorReturn
  style/useTemplate
  correctness/noUnusedLabels
  complexity/noUselessTernary
  correctness/noUnreachableSuper
  suspicious/noExplicitAny
  suspicious/noCompareNegZero
  correctness/noSwitchDeclarations
  a11y/noAutofocus
  correctness/noUnsafeOptionalChaining
  correctness/noConstAssign
  suspicious/noControlCharactersInRegex
  complexity/noUselessTypeConstraint
  style/noVar
  suspicious/noDoubleEquals
  suspicious/noMisleadingCharacterClass
  correctness/noPrecisionLoss
  a11y/noLabelWithoutControl
  style/useDefaultSwitchClause
  suspicious/noRedeclare
  correctness/noSetterReturn
  correctness/noInvalidConstructorSuper
  suspicious/noDuplicateObjectKeys
  suspicious/noFallthroughSwitchClause
  a11y/useKeyWithClickEvents
  suspicious/noUnsafeDeclarationMerging
  complexity/noUselessThisAlias
  correctness/noUnreachable
  correctness/noInnerDeclarations
  suspicious/noDuplicateCase
  style/noParameterAssign
  a11y/useValidAnchor
  complexity/useRegexLiterals
  correctness/noSelfAssign
  style/noUselessElse
  suspicious/noEmptyBlockStatements
  suspicious/noShadowRestrictedNames
  a11y/useMediaCaption
  complexity/noUselessLabel
  complexity/noUselessCatch
  correctness/noUnsafeFinally
  a11y/useAriaPropsForRole
  style/useCollapsedElseIf
  correctness/noNonoctalDecimalEscape
  a11y/useHtmlLang
  complexity/noUselessUndefinedInitialization
  suspicious/noConsole
  style/useNumericLiterals
  a11y/noNoninteractiveTabindex
  correctness/useYield
  suspicious/noImportAssign
  a11y/noInteractiveElementToNoninteractiveRole
  suspicious/useDefaultSwitchClauseLast
  suspicious/noLabelVar
  correctness/noGlobalObjectCalls
  a11y/useAltText
  correctness/noEmptyCharacterClassInRegex
  suspicious/noSparseArray
  a11y/useIframeTitle
  complexity/noBannedTypes
  correctness/noVoidElementsWithChildren
  style/useThrowOnlyError
  style/useAsConstAssertion
  suspicious/noPrototypeBuiltins
  suspicious/noMisleadingInstantiator
  complexity/noUselessLoneBlockStatements
  complexity/noVoid
  a11y/useValidAriaValues
  style/noArguments
  suspicious/noDebugger
  a11y/useFocusableInteractive
  suspicious/noGlobalAssign
  suspicious/noCommentText
  suspicious/useGetterReturn
  style/useFragmentSyntax
  a11y/noPositiveTabindex
  correctness/noEmptyPattern
  security/noDangerouslySetInnerHtmlWithChildren
  suspicious/noDuplicateJsxProps
  a11y/useKeyWithMouseEvents
  suspicious/noExtraNonNullAssertion
  correctness/useExhaustiveDependencies
  security/noGlobalEval
  style/noYodaExpression
  a11y/noRedundantRoles
  style/useConst
  correctness/useHookAtTopLevel
  correctness/noUnusedVariables
  correctness/useIsNan
  suspicious/noSelfCompare
  style/useConsistentBuiltinInstantiation
  suspicious/noAsyncPromiseExecutor
  security/noDangerouslySetInnerHtml
  a11y/noDistractingElements
  suspicious/noArrayIndexKey
  complexity/noWith
  suspicious/noDuplicateClassMembers
  complexity/noExtraBooleanCast
  a11y/useValidAriaProps
  a11y/noRedundantAlt
  correctness/noChildrenProp
  suspicious/noConfusingLabels
  a11y/useButtonType
  a11y/noAriaUnsupportedElements
  style/useShorthandAssign
  a11y/noBlankTarget
  a11y/useHeadingContent
  correctness/useValidForDirection
  correctness/noInvalidUseBeforeDeclaration
  a11y/useAnchorContent
  complexity/noUselessRename
  complexity/noUselessConstructor
  a11y/noAccessKey
  style/noRestrictedGlobals
  style/useExponentiationOperator
  suspicious/useAwait
  correctness/noUndeclaredVariables
  style/useSingleVarDeclarator
  a11y/noNoninteractiveElementToInteractiveRole
  style/noCommaOperator
  style/noNamespace
  a11y/noHeaderScope
  complexity/noUselessFragments
  correctness/noNewSymbol
  complexity/noUselessStringConcat
  suspicious/noClassAssign
  style/noImplicitBoolean
  suspicious/noFunctionAssign

Workspace:
  Open Documents:               0

```
```


### Rule name

Not a specific rule

### Playground link

n/a

### Expected result

It should not throw an error

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Please provide a reproduction

**Comment by arendjr** [comment]
It sounds related to, possibly a duplicate of https://github.com/biomejs/biome/issues/4181
